src/gdk/gdk_bbp.c

```diff
@@ -1518,7 +1517,7 @@ BBPexit(void)
 	/* free all memory (just for leak-checking in Purify) */
 	do {
 		skipped = 0;
-		for (i = 0; i < (bat) ATOMIC_GET(BBPsize, BBPsizeLock); i++) {
+		for (i = 0; i < (bat) BBPsize; i++) {
```

Comment:
I suppose there is a reason for this lock. Why remove?

Reply:
I am locking the whole function. I was getting a race condition while shutting down the embedded database in one of the stress tests, so I had to lock the whole of it.